### PR TITLE
Adding the Requirements doc to repo 

### LINKED ISSUE #8: Install and Document Required Python Libraries
**Objective**: Install necessary Python libraries (e.g., pylsl, numpy, matplotlib) and update the requirements file
**Steps**
- [x] Install libraries (`pip install pylsl numpy matplotlib`).
- [x] Generate a requirements.txt file (`pip freeze > requirements.txt`).
- [x] Add instructions in the `README `on how to install dependencies (`pip install -r requirements.txt`)

### PR DESCRIPTION
**Pull request by JackMcPhillips1543**
Fixes #8 
- Installed the required dependencies:
- Saved dependencies to a .txt file for users to download

All of these changes were required because the repo was empty and we needed to get all the requirements

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
+aiohttp==3.9.1
```
Comment (colinseper) [inline backend/requirements.txt:1]: Tried running the commands on my local machine with a Virtual Enviornment and had less dependencies in my requirements.txt file (using pip install -r requirements.txt failed). When I did a pip freeze these were the requirements I got were in the attached requirements file. 
[requirements.txt](https://github.com/user-attachments/files/17020836/requirements.txt)

**Review by colinseper — CHANGES_REQUESTED** [review]
Tried running the commands on my local machine with a Virtual Enviornment and had less dependencies in my requirements.txt file (using pip install -r requirements.txt failed). When I did a pip freeze these were the requirements I got were in the attached requirements file. 
[requirements.txt](https://github.com/user-attachments/files/17020836/requirements.txt)

Additionally, the README is missing instructions on installing dependencies.
Task #8: Add instructions in the README on how to install dependencies (pip install -r requirements.txt)

**Comment by JackMcPhillips1543** [comment]
> Tried running the commands on my local machine with a Virtual Enviornment and had less dependencies in my requirements.txt file (using pip install -r requirements.txt failed). When I did a pip freeze these were the requirements I got were in the attached requirements file. [requirements.txt](https://github.com/user-attachments/files/17020836/requirements.txt)
> 
> Additionally, the README is missing instructions on installing dependencies. Task #8: Add instructions in the README on how to install dependencies (pip install -r requirements.txt)

I fixed the read me, can you try to run it again once you cd into the backend folder

**Comment by kate-holdener** [comment]
> * Installed the required dependencies:
> * Saved dependencies to a .txt file for users to download
> * Closes Issue [Install and Document Required Python Libraries #8](https://github.com/oss-slu/Step_Time_Biofeedback/issues/8)
> 
> All of these changes were required because the repo was empty and we needed to get all the requirements

@JackMcPhillips1543 Please use:
Fixes #8 
At the top of the PR (do this for future PRs too)

File: README.md
Comment (kate-holdener) [inline README.md:1]: It would be also good to specify instructions for creating a virtual environment in python

**Review by git-voo — CHANGES_REQUESTED** [review]
Good job @JackMcPhillips1543! Your branch name and commit message follow our accepted standard. I like that.  I have just a few things for you to note and/or correct.

1. I see about 59 dependencies in your requirements.txt, we don't need all of this yet, right? we will install them as we go let's not crowd the list with redundant dependencies. Work with @Nissa9902 and @colinseper to learn of the dependencies they made use of and restrict your list to just those.
2. For future pull requests, only contributions that directly modify the codebase will be considered for merging into the main branch. An exception will be made for this PR because the requirement for code-based contributions was not communicated by me early enough.
3. Remember to always add me as a reviewer to your PRs. By doing this, I will get a notification that facilitates how promptly you get feedback.
Always assign the PR to yourself for the same reason as above. It helps us keep track, know who to reach out to for what, get notified promptly, and maintain professionalism.

**Review by colinseper — DISMISSED** [review]
The updates all look good on my end, nice job!

**Review by git-voo — DISMISSED** [review]
Good. we can work this this

**Review by git-voo — APPROVED** [review]

**Pull request merged**